app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/templates.tsx

```diff
@@ -1,62 +1,137 @@
+const genBackgroundImage = () => `url(${Math.random() > 0.5 ? dataUrl1 : dataUrl2})`
 
-  return (
-    <div className="bg-white">
-      <div className="mx-auto py-16 px-4 sm:py-24 sm:px-6 lg:px-8">
-        <div className="grid grid-cols-1 gap-y-10 gap-x-6 sm:grid-cols-3 lg:grid-cols-4 xl:grid-cols-5 xl:gap-x-8">
-          {templates.map((template, index) => (
-            <div
-              key={index}
-              className="h-40 rounded-lg border-4 border-dashed border-gray-200 flex"
-            >
-              <div className="m-auto text-center">
-                <button
-                  onClick={() => {
-                    onChange(template.templateName);
-                  }}
+/** Render single template */
+interface TemplateRenderProps extends TemplatesProps {
+    template: Template
+}
+// eslint-disable-next-line @typescript-eslint/naming-convention
+const TemplateRender: React.FC<TemplateRenderProps> = ({ template, onChange }) => {
+    /** Unify the border color and the text color */
+    const borderColor = '9E8C91'
+    /**
+     * Dash border spacing is not supported by native CSS.
+     * Therefore, use a background image to create the border.
+     * It is essentially an SVG image that was generated by the website.
+     * @see {@link https://kovart.github.io/dashed-border-generator}
+     */
+    const borderBgImg = `url("data:image/svg+xml,%3csvg width='100%25' height='100%25' xmlns='http://www.w3.org/2000/svg'%3e%3crect width='100%25' height='100%25' fill='none' rx='16' ry='16' stroke='%23${borderColor}FF' stroke-width='4' stroke-dasharray='15%2c 15' stroke-dashoffset='0' stroke-linecap='butt'/%3e%3c/svg%3e")`
+    return (
+        <button
+            className="h-40 cursor-pointer"
+            onClick={() => {
+                onChange(template.templateName)
+            }}
+        >
+            {template.templateName ? (
+                <div
+                    style={{ backgroundImage: genBackgroundImage() }}
+                    className="flex flex-col justify-end h-full w-full rounded-[16px] overflow-hidden text-white text-left bg-cover"
```

Comment:
not sure using computed values is a good idea - is there a reason you're not using the normal tailwind classes? (e.g. `rounded-2xl`: <https://tailwindcss.com/docs/border-radius#rounded-corners>)